packages/db/src/translate/granules.ts

```diff
@@ -1,11 +1,93 @@
 
 import { CollectionPgModel } from '../models/collection';
 import { PdrPgModel } from '../models/pdr';
-import { PostgresGranule } from '../types/granule';
+import { PostgresGranule, PostgresGranuleRecord } from '../types/granule';
 import { ProviderPgModel } from '../models/provider';
+import { FilePgModel } from '../models/file';
+import { translatePostgresFileToApiFile } from './file';
+import { getExecutionArnsByGranuleCumulusId } from '../lib/execution';
+
+/**
+ * Generate an API Granule object from a Postgres Granule with associated Files.
+ *
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} granulePgRecord
+ *   Granule from Postgres
+ * @param {Knex | Knex.Transaction} knexOrTransaction
+ *   Knex client for reading from RDS database
+ * @param {Object} collectionPgModel - Instance of the collection database model
+ * @param {Object} pdrPgModel - Instance of the pdr database model
+ * @param {Object} providerPgModel - Instance of the provider database model
+ * @param {Object} filePgModel - Instance of the file database model
+ * @param {Object} executionPgModel - Instance of the execution database model
+ * @returns {Object} An API Granule with associated Files
+ */
+export const translatePostgresGranuleToApiGranule = async (
+  granulePgRecord: PostgresGranuleRecord,
+  knexOrTransaction: Knex | Knex.Transaction,
+  collectionPgModel = new CollectionPgModel(),
+  pdrPgModel = new PdrPgModel(),
+  providerPgModel = new ProviderPgModel(),
+  filePgModel = new FilePgModel()
+): Promise<AWS.DynamoDB.DocumentClient.AttributeMap> => {
+  const collection = await collectionPgModel.get(
+    knexOrTransaction, { cumulus_id: granulePgRecord.collection_cumulus_id }
+  );
+  const files = await filePgModel.search(
+    knexOrTransaction, { granule_cumulus_id: granulePgRecord.cumulus_id }
+  );
+  const executionArns = await getExecutionArnsByGranuleCumulusId(
+    knexOrTransaction,
+    granulePgRecord.cumulus_id
+  );
+
+  let pdr;
+
+  if (granulePgRecord.pdr_cumulus_id) {
+    pdr = await pdrPgModel.get(
+      knexOrTransaction, { cumulus_id: granulePgRecord.pdr_cumulus_id }
+    );
+  }
+
+  let provider;
+
+  if (granulePgRecord.provider_cumulus_id) {
+    provider = await providerPgModel.get(
+      knexOrTransaction, { cumulus_id: granulePgRecord.provider_cumulus_id }
+    );
+  }
+
+  return removeNilProperties(({
+    granuleId: granulePgRecord.granule_id,
+    status: granulePgRecord.status,
+    collectionId: constructCollectionId(collection.name, collection.version),
+    published: granulePgRecord.published,
+    duration: granulePgRecord.duration,
+    timeToArchive: granulePgRecord.time_to_archive,
+    timeToPreprocess: granulePgRecord.time_to_process,
+    productVolume: granulePgRecord.product_volume
+      ? Number.parseInt(granulePgRecord.product_volume, 10) : undefined,
+    error: granulePgRecord.error,
+    cmrLink: granulePgRecord.cmr_link,
+    pdrName: pdr ? pdr.name : undefined,
```

Comment:
Nit: Consider setting provider/pdr/etc to `{}` and updating this to pdr?.name and so forth.

Reply:
Ah yeah, forgot that syntax existed.

https://github.com/nasa/cumulus/pull/2415/commits/b97677777f2b5e484732eb4e68d382d99c1d708b

I'm not sure I followed the first part though "Consider setting provider/pdr/etc to {}". Are you saying `let provider = {};`? I don't think we can do that because we'll get a `property 'name' does not exist on {}` below. We could make it a `PostgresProviderRecord` in that initial declaration. Or something? I think I'm just misunderstanding.

Reply:
No, I'm just being dumb.   Because provider is `undefined | <result of providerPgModel.get` we *need* the ternary.       You should revert that change and ignore the above comment, apologies 😓

Reply:
Reverting, and again, apologies 1f4795464